README.md

```diff
@@ -194,54 +199,46 @@ An [Effect] are the instruction to the [Executor] to run set of [Task]s concurre
 ## 2.3 IPLD Schema
 
 ```ipldsch
-type Task struct {
-  on      URI
-  call    Ability
-  input   {String : Any}
-
-  nnc     string
+type Instruction struct {
+  rsc    URI
+  op     Ability
+  input  {String : Any}
+  nnc    String
 }
 
 type URI string
 type Ability string
 
+type Task struct {
+  run     &Instruction
+  meta    {String : Any}
+  prf     [&UCAN]
+
+  # Receipt of the invocation that caused this Task to be run
+  cause   optional &Receipt
+}
+
 type Authorization struct {
-  # Authorization is denoted by the set of links been authorized
-  scope   [&Any]
-  # Scope signed by the invoker
-  s       VarSig
+  scope   [&Any] # The set of authorized links
+  s       Varsig # Scope signed by the invoker
 }
 
 type Invocation struct {
-  v       SemVer
-
-  run     &Task
-
-  # Receipt of the invocation that caused this invocation
-  cause   optional &Invocation
-
-  # Task authorization.
+  task    Task
   auth    &Authorization
-
-  meta    {String : Any}
-
-  prf     [&UCAN]
 }
 
-type SemVer string
+type InvocationTag union {
+  | Invocation   "ucan/invoke@0.2.0"
+} representation keyed
 
 type Receipt struct {
-  # Invocation this is a receipt for
-  ran     &Invocation
+  ran     &Invocation # Invocation this is a receipt for
 
-  # Output of the invocation
-  out     Result
+  out     Result # Output of the invocation
+  fx      Effects # Effects to be enqueued
 
-  # Effects to be performed
-  fx      Effects
- 
-  # All the other metadata
-  meta    {String : Any}
+  meta    {String : Any} # All the other metadata
 
   # Principal that issued this receipt. If omitted issuer is
```

Comment:
going off the Ucan spec, is this not `Principle`?

Reply:
Oh, that would be a typo iun the UCAN spec then 😅 For example, from the JWT spec:

> The "iss" (issuer) claim identifies the principal that issued the JWT.

Reply:
The UCAN spec actually uses both senses of principle/principal, which can be confusing:

> Following the principle of least authority

...and...

> Agent | The general class of entities and principals that interact with a UCAN

Reply:
ah yes. rs-ucan uses `Principle` (in the code) haha.

Reply:
Oh LOL we should probably fix that